kernel/src/syscall/execve.rs

```diff
@@ -118,12 +119,40 @@ fn do_execve(
     drop(closed_files);
 
     debug!("load program to root vmar");
+
+    // FIXME: Currently, the efficiency of replacing the VMAR is lower than that
+    // of directly clearing the VMAR. Therefore, we only replace the VMAR in the
+    // case of vfork here.
+    let new_vmar = if process.status().is_vfork_child() {
+        let new_vmar = Vmar::<Full>::new_root();
+        *ctx.thread_local.root_vmar().borrow_mut() = Some(new_vmar.dup().unwrap());
```

Comment:
We need to disable preemption when updating `thread_local.root_vmar()`. Moreover, the code should not live in `sys_execve` because it's a basic operating principle rather than system call specific logic.